core/src/main/java/com/scalar/db/common/checker/OperationChecker.java

```diff
@@ -357,6 +366,72 @@ public void check(List<? extends Mutation> mutations) throws ExecutionException
     }
   }
 
+  private boolean isOutOfAtomicityUnit(
+      Mutation mutation1, StorageInfo storageInfo1, Mutation mutation2) throws ExecutionException {
+    assert mutation1.forNamespace().isPresent()
+        && mutation1.forTable().isPresent()
+        && mutation2.forNamespace().isPresent()
+        && mutation2.forTable().isPresent();
+
+    switch (storageInfo1.getMutationAtomicityUnit()) {
+      case RECORD:
+        if (!mutation1.getClusteringKey().equals(mutation2.getClusteringKey())) {
```

Comment:
I might be missing something, but this seems only checking the clustering keys. Is it possible the 2 mutations have different namespace, table, or partition keys here?